### ISSUE TITLE
BUG (?) -> Theres no option to train the classifier withou augmentation

### ISSUE THREAD

**Issue by VitorBonella** [silvermete0r/oikan#53]
`python
class OIKANClassifier(OIKAN):
    """OIKAN model for classification tasks."""
    def fit(self, X, y):
        """
        Fits the classifier to the data.
        
        Parameters:
        -----------
        X : array-like of shape (n_samples, n_features)
            Training data.
        y : array-like of shape (n_samples,)
            Target labels.
        """
        X = np.asarray(X)
        from sklearn.preprocessing import LabelEncoder
        le = LabelEncoder()
        y_encoded = le.fit_transform(y)
        self.classes_ = le.classes_
        n_classes = len(self.classes_)
        y_onehot = nn.functional.one_hot(torch.tensor(y_encoded), num_classes=n_classes).float()
        self._train_neural_net(X, y_onehot, output_size=n_classes, loss_fn=nn.CrossEntropyLoss())
        if self.augmentation_factor > 1:
            if self.verbose:
                print(f"Original data: features shape: {X.shape} | target shape: {y.shape}")
            
            X_aug = self._generate_augmented_data(X)
            
            self.neural_net.eval()
            with torch.no_grad():
                logits_aug = self.neural_net(torch.tensor(X_aug, dtype=torch.float32)).detach().numpy()
            
            if self.verbose:
                print(f"Augmented data: features shape: {X_aug.shape} | target shape: {logits_aug.shape}")
            
            X_combined = np.vstack([X, X_aug])
            y_combined = np.vstack([y_onehot.numpy(), logits_aug])
        else:
            if self.verbose:
                print("Skipping neural network training (augmentation_factor=1)")
            X_combined = X
            y_combined = y_onehot.numpy()
            
        self._perform_symbolic_regression(X_combined, y_combined)
        if self.verbose:
            print("OIKANClassifier model training completed successfully!")`

changing the line   self._train_neural_net(X, y_onehot, output_size=n_classes, loss_fn=nn.CrossEntropyLoss()) out of the condition resolves it.
